### PR TITLE
RadioTap flags support + Dot11 FCS&improvements

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- fixes https://github.com/secdev/scapy/issues/1379 by improving RadioTap fields support. This required the implementation of ReversePadField (RadioTap padding is reversed), which itself required a `pkt._tmp_dissect_pos`)
- Supports FCS in Dot11
- introducing `pkt._tmp_dissect_pos`. It is a variable which contains the size of the current layer, at the time that it is used, and can only be used during the dissection. It is very useful in `length_from` functions, as it gets what has been exactly dissected when the lambda gets executed. It is changing dynamically
- merging contrib/ppi with dot11: the `PPI()` layer was implemented twice.

### CODE REVIEW

**Comment by calebmadrigal** [comment]
This seems to be working to me (tested on both Ubuntu Linux and macOS 10.13). Any idea when we can get scapy 2.4.1? I'd love to use this as soon as possible in my project (https://github.com/calebmadrigal/trackerjacker).

File: scapy/layers/dot11.py [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Scapy. If not, see <http://www.gnu.org/licenses/>.
+
 # Copyright (C) Philippe Biondi <phil@secdev.org>
-# This program is published under a GPLv2 license
+# PPI author: <jellch@harris.com>
```
Comment (guedou) [inline scapy/layers/dot11.py]: Original PPI author?

**Comment by guedou** [comment]
I restarted the failing test.

File: scapy/layers/dot11.py (hunk 1 above)
Comment (gpotter2) [reply scapy/layers/dot11.py]: From the contributor modules

File: scapy/contrib/ppi_geotag.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 import time
 from scapy.packet import *
 from scapy.fields import *
-from scapy.contrib.ppi import PPIGenericFldHdr, addPPIType
+from scapy.layers.dot11 import *
```
Comment (p-l-) [inline scapy/contrib/ppi_geotag.py]: Could you limit what is imported here?

**Comment by gpotter2** [comment]
@p-l- Updated: moved `PPI `to `ppi.py`

**Comment by p-l-** [comment]
Can you rebase against current master? We have a conflict due to removed tests.

**Comment by guedou** [comment]
@gpotter2 this PR conflicts.

**Comment by gpotter2** [comment]
Fixed conflicts

**Comment by calebmadrigal** [comment]
@gpotter2 Any ETA on a 2.4.1 release? It'd be great prior to DEF CON 2018! :)

**Comment by gpotter2** [comment]
Oh, you’re not the first one wanting 2.4.1 for DEF CON :) I have created a project a few days ago, but I doubt it will be ready for August...

Why do you need 2.4.1 ? Can’t you just use the dev build ? You can even install it via pip with git (lookup)

**Comment by calebmadrigal** [comment]
@gpotter2 I basically just need the stuff in this PR. I could use a dev build for personal use, but I'd like a nice clean pip "scapy==2.4.1" for dependency management.

**Comment by gpotter2** [comment]
We still have stuff to check :/
https://github.com/secdev/scapy/projects/2

**Comment by guedou** [comment]
I am not sure that v2.4.1 will be released before the conference =\
If we cannot do it, you can try to specify a git commit in requirements.txt

**Pull request merged**